tests/datasource/fluent/test_config.py

```diff
@@ -283,6 +292,48 @@ def test_from_gx_config(self, asset_dict: dict):
         )
 
 
+def test_id_only_serialized_if_present(ds_dict_config: dict):
```

Comment:
Why are we allowing `id` to be `Optional`?

Reply:
Because nothing relies on it.
Cloud will generate IDs for the objects, but that will be done server-side, not by the client.